File: codes/seive/spoj-TDKPRIME.cpp
#include<bits/stdc++.h>
using namespace std;

int max_size=90000001;
bool l[90000001];
vector<int>k_prime;

void seive(){
    l[0]=true;
    l[1]=true;
    for(int i=2;i*i<max_size;i++){
        if(l[i]==false){
            for(int j=i*i;j<max_size;j=j+i){
                l[j]=true;
            }
        }
    }

}
void get_k_prime_list(){
    for(int i=1;i<max_size;i++){
        if(l[i]==false){
            k_prime.push_back(i);
        }
    }

}

int main(){
seive();
get_k_prime_list();
int cases;
cin>>cases;
for(int i=0;i<cases;i++){
    int k;
    cin>>k;
    cout<<k_prime[k-1]<<endl;
}

}

File: codes/seive/spoj-TDKPRIME.py
size=90000000
l=[True]*(size+1)
k_prime=[]
def seive(n):
    l[0]=False
    l[1]=False
    i=2
    while i*i<n:
        if l[i]==True:
            for j in range(i*i,n,i):
                l[j]=False
        i=i+1
def get_k_prime_array():
    for i in range(1,size+1):
        if l[i]==True:
            k_prime.append(i)




seive(size)
get_k_prime_array()
cases=int(input())
for i in range(cases):
    k=int(input())
    print(k_prime[k-1])
    

